Dockerfile

```diff
@@ -224,3 +224,7 @@ COPY --from=static /opt/warehouse/src/warehouse/static/dist/ /opt/warehouse/src/
 COPY --from=static /opt/warehouse/src/warehouse/admin/static/dist/ /opt/warehouse/src/warehouse/admin/static/dist/
 COPY --from=build /opt/warehouse/ /opt/warehouse/
 COPY . /opt/warehouse/src/
+
+RUN adduser warehouse
+RUN chown -R warehouse /opt/warehouse/src
+USER warehouse
```

Comment:
This is the one thing I'm not the most certain about, since it changes real runtime behavior to execute all processes under this user, vs previously implict `root`.

It's a requirement for postgres, as pg's built-in logic will actively deny it from being started as `root` for safety reasons.

Reply:
cabotage currently forces built containers to `nobody` [here](https://github.com/cabotage/cabotage-app/blob/2d7c0262ec347734adfde86e859a2a27b18e3964/cabotage/utils/release_build_context.py#L3-L10).

i'm not sure how using a named user like `warehouse` will interact.

i recommend remove the `adducer` and `chown` parts and just replacing it with `USER nobody`

Reply:
I tried replacing, and it failed immediately, since it cannot write coverage.

Considering that cabotage adds the last layer of nobody and sets an entrypoint, and we need the ability to write local files for coverage, or creating db revisions during development, I'm electing to keep this user as is, and add a comment above.